app/services/csv_exporter.rb

```diff
@@ -4,13 +4,17 @@ module CsvExporter
   def self.export(resources, columns, header = nil)
     header ||= default_header(columns)
     raise ArgumentError, "Columns and header row aren't the same length" unless columns.length == header.length
+    # we need to pass the current user to the enumerator which is executed by a seperate thread
```

Comment:
the resources are correctly scoped actually, it's only when loading relations that this fails as there is no current user inside the enumerator.